src/vs/workbench/contrib/interactive/browser/interactiveWindowFileSystem.ts

```diff
@@ -0,0 +1,76 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { Emitter, Event } from 'vs/base/common/event';
+import { Disposable, IDisposable } from 'vs/base/common/lifecycle';
+import { URI } from 'vs/base/common/uri';
+import { FileSystemProviderCapabilities, FileType, IFileChange, IFileDeleteOptions, IFileOpenOptions, IFileOverwriteOptions, IFileSystemProvider, IStat, IWatchOptions } from 'vs/platform/files/common/files';
+
+export class InteractiveWindowFileSystem implements IFileSystemProvider {
```

Comment:
Maybe you can reduce the complexity of this class by extending the `InMemoryFileSystemProvider` and simply overriding the methods that write to the memory by skipping the write. This will make this a very slim class in the end.